### PR TITLE
Make rust binding really useful

### PR DESCRIPTION
**Pull request by liuchengxu**
Perhaps I'm the first one to use the Rust binding of tree-sitter-vim :P This commit uncomments the available queries so that it is really usable.

`HIGHLIGHT_QUERY` follows the convention of other Rust bindings.

### CODE REVIEW

**Comment by clason** [comment]
Thank you, but this doesn't make the bindings useful since the queries are written specifically for Neovim and are not usable as-is outside that.

**Comment by liuchengxu** [comment]
Actually, I'm using it to do the syntax highlighting in my plugin, which works in both Vim and Neovim. Take it look [here](https://github.com/liuchengxu/vim-clap/tree/tree-sitter-highlight) if you are interested.

**Comment by liuchengxu** [comment]
I came to this repo when I first searched tree sitter vim online, importing to my dependencies, but to find these queries are not exported, which is annoying. I'm fine with maintaining my own fork as the changes are minimal, but I think this is a generally useful improvement that might be beneficial to others in the future.

**Comment by clason** [comment]
The issue is that these bindings are generated, so even if we did do this, we need to do this in a way that's maintainable (i.e., not have to do this manually every time we need to regenerate them due to a tree-sitter bump).

**Comment by liuchengxu** [comment]
These generated content are super stable, there is no need to regenerate them in the foreseeable future unless an exception from what I see. It shouldn't be hard to follow even if there are breaking changes upstream, I'm happy to maintain the Rust binding part if you are concerned since I'm using it. I suppose we're willing to support the Rust binding seriously since it's generated and already there, right? But it's not really usable right now.

**Comment by clason** [comment]
>  I suppose we're willing to support the Rust binding seriously since it's generated and already there, right?

No, that's exactly the point: so far we are _not_ willing to make that support guarantee; we only guarantee support for the binary parser and queries in Neovim. (The bindings are there "by mistake"; they were never added deliberately and would not have been, if `tree-sitter generate` did not have the -- controversial -- feature of generating them unasked.)

We can revisit this after the next tree-sitter release, which does make some major changes.

**Comment by liuchengxu** [comment]
>so far we are not willing to make that support guarantee

Okay, that's fair, nice to have it documented somewhere (maybe adding a README in the Rust binding) to avoid potential confusion.

**Comment by clason** [comment]
Let's wait for the next release and re-evaluate. This is not a situation unique to tree-sitter-vim and applies to all parsers. (The preferred alternative -- if we don't maintain them --  is to _remove_ the bindings and tell people to just generate them themselves.)
